static/js/message_events.js

```diff
@@ -361,13 +361,22 @@ export function update_messages(events) {
                     const predicate = current_filter.predicate();
                     message_ids_to_remove = event_messages.filter((msg) => !predicate(msg));
                     message_ids_to_remove = message_ids_to_remove.map((msg) => msg.id);
+                    // We filter out messages that do not belong to the message
+                    // list and then pass these to the remove messages codepath.
+                    // While we can pass all our messages to the add messages
+                    // codepath as the filtering is done within the method.
+                    message_lists.current.remove_and_rerender(message_ids_to_remove);
+                    message_lists.current.add_messages(event_messages);
+                } else if (!current_filter) {
+                    message_lists.current.add_messages(event_messages);
```

Comment:
This block doesn't seem likely to be correct.  I think this case applies to the home view -- it should already contain the relevant messages in most cases. 
 
I suppose the cases where it wouldn't is if the messages were moved from a private stream the current user is not subscribed to while you're looking at the home view.  I would expect this logic to not be correct for that case either; what we'd want is something that ALSO updates the home/All messages view unconditionally (i.e. outside this if/else clause), since we leave it rendered even when viewing other views.  Does this block fix a bug you can reproduce at present?

If not, we might want to skip this block and leave it for a follow-up.